packages/cubejs-api-gateway/src/helpers/sqlQueryHandler.ts

```diff
@@ -0,0 +1,80 @@
+export enum SQLQueryType {
+  Select = 'select',
+  With = 'with',
+  Create = 'create',
+  Alter = 'alter',
+  Drop = 'drop',
+  Truncate = 'truncate',
+  Insert = 'insert',
+  Update = 'update',
+  Delete = 'delete',
+  Explain = 'explain',
+}
+
+interface SQLRunnerQueryType {
+  type: SQLQueryType;
+  regex?: RegExp;
+  shouldAddLimit?: boolean;
+}
+
+function createSQLRunnerQueryType({
+  type,
+}: SQLRunnerQueryType): SQLRunnerQueryType {
+  const regex: RegExp = new RegExp(`^(${type})\\b`, 'i');
+  let shouldAddLimit: boolean = false;
+
+  if (type === SQLQueryType.Select || type === SQLQueryType.With) {
+    shouldAddLimit = true;
+  }
+
+  return {
+    type,
+    regex,
+    shouldAddLimit,
+  };
+}
+
+export function getSQLRunnerQueryType(sql: string): SQLRunnerQueryType | null {
```

Comment:
Actually, you can rename it to `isRequiredLimit` and return boolean value from it.